system/security/securesocket/securesocket.cpp

```diff
@@ -896,6 +900,9 @@ class CSecureSocketContext : implements ISecureSocketContext, public CInterface
         {
             throw MakeStringException(-1, "ctx can't be created");
         }
+
+        setSessionIdContext();
```

Comment:
I think setting the ctx session is meaningful only on the server side ?
If that is true, should we do this only for server, ie:
```
        if(sockettype == ServerSocket)
            setSessionIdContext();
```